src/passwordValidation.js

```diff
@@ -11,16 +11,19 @@ export default function isValidPassword(password = "") {
   if (typeof password !== "string") password = String(password);
   if(password.length!==10)return false    //less than 10
 
-  if(!password.match(/\w+^[_]/))return false    //happy case
+  // if(!password.match(/\w+^[_]/))return false    //happy case
 
-  if(!password.match(/\d+/))return false   //has to have numbers
+  if(!(/\d+/).test(password))return false   //has to have numbers
 
+  if(!(/[a-z]/i).test(password))return false  //has to have letters
 
-  if(!password.match(/[a-zA-Z]/))return false     //has to have letters
+  if((/[!@#$%^&*(),.?":{}|<>]/).test(password))return false //must not have special characters
 
+  if(!(/[A-Z]/).test(password)) return false  // must have upper case letters
```

Comment:
It would be better to check if it contain capital and small letters in one if condition and one regex expression because all of these are some how close related

Reply:
Even numbers